File: packages/core/types/template/PreprocessorDirectives.ts
import { IBaseNode, TemplateNodeTypes } from "./BaseNode/BaseTypes";

export interface IIncludeDirective extends IBaseNode {
  nodeType: TemplateNodeTypes.IncludeDirective;
  name: string;
}

export interface IMacroDefinition extends IBaseNode {
  nodeType: TemplateNodeTypes.MacroDefinition;
}

File: packages/ssat/src/ssat/types/template/preprocessor_directives.py
"""Preprocessor directive node type definitions."""

from .BaseNode.base_types import IBaseNode, TemplateNodeTypes


class IIncludeDirective(IBaseNode):
    """Include directive node."""

    nodeType: TemplateNodeTypes = TemplateNodeTypes.IncludeDirective
    name: str


class IMacroDefinition(IBaseNode):
    """Macro definition node."""

    nodeType: TemplateNodeTypes = TemplateNodeTypes.MacroDefinition


